rust_dev_preview/Dockerfile

```diff
@@ -1,9 +1,11 @@
 # syntax=docker/dockerfile:1
 # Status: Beta
 # GA updates: https://github.com/awsdocs/aws-doc-sdk-examples/issues/4134
-ARG MSRV=1.58
+ARG MSRV=1.62.1
 FROM rust:$MSRV
 COPY . /rust_dev_preview/
-RUN cd /rust_dev_preview && \
-    cargo build --ignore-rust-version
+WORKDIR /rust_dev_preview
+RUN apt-get update && \
+  apt-get install libclang-dev -y && \
+  cargo check
```

Comment:
`WORKDIR` will fail in GitHub but pass locally due to [this issue](https://docs.github.com/en/actions/creating-actions/dockerfile-support-for-github-actions#workdir).
```suggestion
RUN cd /rust_dev_preview && \
  apt-get update && \
  apt-get install libclang-dev -y && \
  cargo check
```

Reply:
Ooh that's nasty...
 Updated with a rust idiomatic solution. WORKDIR is still important for non-actions users.